oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentPropertyState.java

```diff
@@ -128,7 +188,7 @@ public boolean equals(Object object) {
         } else if (object instanceof DocumentPropertyState) {
             DocumentPropertyState other = (DocumentPropertyState) object;
             return this.name.equals(other.name)
-                    && this.value.equals(other.value);
+                    && this.getValue().equals(other.getValue());
```

Comment:
Hm. That means the "equals()" method would need to uncompress. That seems performance critical.

Reply:
Yes.

Reply:
is there a way that equals wouldn't have to uncompress?

Reply:
We could special-case the compressed case, and compare the byte arrays. Still expensive.

Maybe we could keep the String's hashCode in order to avoid comparing the byte arrays in most cases.

Reply:
Right, I think there should be distinction between non-compressed, compressed, mixed cases. Only the compressed one is slightly more involved. For that one we could use a combination of length comparison, hashcode (calculated when needed) and/or plain Arrays.equals(byte[],byte[])..

Reply:
Done